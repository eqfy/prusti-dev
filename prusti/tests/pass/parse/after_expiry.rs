// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zskip-verify -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]

use prusti_contracts::*;

#[after_expiry_if(a => a, a)]
fn test1() {}

#[after_expiry_if(a, a)]
fn test2() {}

#[after_expiry(a)]
fn test3() {}

#[after_expiry(a => a)]
fn test4() {}

#[after_expiry(
    match x {
        1 => 1,
        2 => 2
    }
)]
fn test5() {}

fn main() {}
