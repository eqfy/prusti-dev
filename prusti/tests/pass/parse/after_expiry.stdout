// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zskip-verify -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]
#[prelude_import]
use std::prelude::v1::*;
#[macro_use]
extern crate std;

use prusti_contracts::*;

fn prusti_pledge_item_test1_$(NUM_UUID)() { }
#[prusti::pledge_spec_id_ref = "$(NUM_UUID)"]
fn test1() { }

fn prusti_pledge_item_test2_$(NUM_UUID)() { }
#[prusti::pledge_spec_id_ref = "$(NUM_UUID)"]
fn test2() { }

fn prusti_pledge_item_test3_$(NUM_UUID)() { }
#[prusti::pledge_spec_id_ref = "$(NUM_UUID)"]
fn test3() { }

fn prusti_pledge_item_test4_$(NUM_UUID)() { }
#[prusti::pledge_spec_id_ref = "$(NUM_UUID)"]
fn test4() { }

fn prusti_pledge_item_test5_$(NUM_UUID)() { }
#[prusti::pledge_spec_id_ref = "$(NUM_UUID)"]
fn test5() { }

fn main() { }
